app/models/test.rb

```diff
@@ -4,7 +4,18 @@ class Test < ApplicationRecord
   has_many :users, through: :users_were_passing_tests
   has_many :users_were_passing_tests
   has_many :questions
-  
+
+  scope :simple, -> { where(level: 0..1) }
+  scope :medium, -> { where(level: 2..4) } 
+  scope :hard, -> { where(level: 5..Float::INFINITY) } 
+  scope :tests_by_category, -> (category) { joins(:category).
+    where(categories: {title: category}).order(title: :DESC).pluck('title')}
+
+  validates :title, presence: true  
```

Comment:
validates :title, presence: true, uniqueness: { scope: :level, message: "A test with this name and level already exists" }